tester/queue.go

```diff
@@ -2,12 +2,27 @@ package tester
 
 import (
 	"sync"
+
+	"github.com/Shopify/sarama"
 )
 
 type message struct {
-	offset int64
-	key    string
-	value  []byte
+	offset  int64
+	key     string
+	value   []byte
+	headers map[string][]byte
+}
+
+func (m *message) SaramaHeaders() []*sarama.RecordHeader {
+	headers := make([]*sarama.RecordHeader, len(m.headers))
+	idx := 0
```

Comment:
will be always 0, which is not intended I guess

Reply:
New commit takes care of that.  If forgot to remove unused code that I had factored into a method on the tester.message struct

Reply:
I guess this comment was meant for the other code comment, where it was actually removed. Here, the variable `idx` is still always zero. Rewriting the unit test to contain multiple header values would fail, because only the first is not-nil.

I guess it should be like:
```go
headers := make([]*sarama.RecordHeader, 0, len(m.headers))
for k, v := range m.headers {
headers = append(headers, &sarama.RecordHeader{
	Key:   []byte(k),
	Value: v,
})
}
```

Reply:
Yes, that would work.  I meant to increment idx in the loop.  I will correct this.

Reply:
yes of course, just proposing an alternative to get rid of the counter variable

Reply:
I agree, much cleaner.  Committed test and fix.